src/test/java/space/obminyashka/items_exchange/end2end/AdvertisementFlowTest.java

```diff
@@ -105,6 +106,17 @@ void findPaginatedAsThumbnails_shouldReturnSpecificAdvertisementTitleDto() throw
                 .andExpect(jsonPath("$.content[0].ownerName").value("admin"));
     }
 
+    @Test
+    @WithMockUser(username = "user")
+    @DataSet("database_init.yml")
+    void findPaginatedByCategoryId_shouldReturnPageResponse() throws Exception {
+        long id = 1;
+        int page = 0;
+        int size = 12;
+        sendUriAndGetResultAction(get(ADV_BY_CATEGORY_ID, id, page, size), status().isOk())
+                .andExpect(jsonPath("$.numberOfElements").value(advertisementRepository.count()));
```

Comment:
It'd be great to validate that the IDs in the response are the same as they were expected.

Reply:
Added